node-graph/nodes/vector/src/vector_nodes.rs

```diff
@@ -1557,127 +1578,147 @@ async fn morph(_: impl Ctx, source: Table<Vector>, #[expose] target: Table<Vecto
 		}
 	}
 
-	let time = time.clamp(0., 1.);
+	// Preserve original graphic table as upstream data so this group layer's nested layers can be edited by the tools.
+	let graphic_table_content = content.clone().into_graphic_table();
 
-	source
-		.into_iter()
-		.zip(target.into_iter())
-		.map(|(source_row, target_row)| {
-			let mut vector = Vector::default();
+	// If the input isn't a Table<Vector>, we convert it into one by flattening any Table<Graphic> content.
+	let content = content.into_flattened_vector_table();
```

Comment:
I think this should be done with the "FlattenVectorNode", the input to this node should be Table<Vector>, and in the future the automatic type conversion can be added

Reply:
I thought about automatic type conversion but decided we probably don't want that, because we can't guarantee that will always be desired. Going from a less pure to more pure form of data is lossy (we throw away non-vector objects, in this case) and opinionated (do we want to flatten deeply or only one level while discarding or merging the geometry of deeper levels?) and will need to vary per node based on what makes sense for each one. This allows the node to easily reuse this functionality if needed, but not let data get destroyed by the type conversion system if it doesn't encode user intent.